### PR TITLE
deps: bump kubernetes-client to 6.4.0

### PR DESCRIPTION
**Pull request by manusa**
## Description

deps: bump kubernetes-client to 6.4.0

This PR includes changes to the `ClusterAccessTest`. The Kubernetes Mock server as become more stable and performant over the last couple of years. Being able to test `ClusterAccess `against a semi-real cluster is easier to maintain than relying on Mocks that need extra-maintenance. In this case, the Client is slowly moving to abstract away explicit KubernetesClient implementations and moving on to SPI provided and more generic ones instead.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
```
maven/mavencentral/io.fabric8/kubernetes-client-api/6.4.0, Apache-2.0, approved, #6580
maven/mavencentral/io.fabric8/kubernetes-client/6.4.0, Apache-2.0 AND (Apache-2.0 AND LicenseRef-scancode-public-domain-disclaimer), restricted, #6570
maven/mavencentral/io.fabric8/kubernetes-httpclient-okhttp/6.4.0, Apache-2.0, approved, #6584
maven/mavencentral/io.fabric8/kubernetes-model-admissionregistration/6.4.0, Apache-2.0, approved, #6603
maven/mavencentral/io.fabric8/kubernetes-model-apiextensions/6.4.0, Apache-2.0, approved, #6569
maven/mavencentral/io.fabric8/kubernetes-model-apps/6.4.0, Apache-2.0, approved, #6581
maven/mavencentral/io.fabric8/kubernetes-model-autoscaling/6.4.0, Apache-2.0, approved, #6598
maven/mavencentral/io.fabric8/kubernetes-model-batch/6.4.0, Apache-2.0, approved, #6600
maven/mavencentral/io.fabric8/kubernetes-model-certificates/6.4.0, Apache-2.0, approved, #6602
maven/mavencentral/io.fabric8/kubernetes-model-common/6.4.0, Apache-2.0, approved, #6574
maven/mavencentral/io.fabric8/kubernetes-model-coordination/6.4.0, Apache-2.0, approved, #6586
maven/mavencentral/io.fabric8/kubernetes-model-core/6.4.0, Apache-2.0, approved, #6604
maven/mavencentral/io.fabric8/kubernetes-model-discovery/6.4.0, Apache-2.0, approved, #6578
maven/mavencentral/io.fabric8/kubernetes-model-events/6.4.0, Apache-2.0, approved, #6607
maven/mavencentral/io.fabric8/kubernetes-model-extensions/6.4.0, Apache-2.0, approved, #6577
maven/mavencentral/io.fabric8/kubernetes-model-flowcontrol/6.4.0, Apache-2.0, approved, #6601
maven/mavencentral/io.fabric8/kubernetes-model-gatewayapi/6.4.0, Apache-2.0, approved, #6599
maven/mavencentral/io.fabric8/kubernetes-model-metrics/6.4.0, Apache-2.0, approved, #6592
maven/mavencentral/io.fabric8/kubernetes-model-networking/6.4.0, Apache-2.0, approved, #6568
maven/mavencentral/io.fabric8/kubernetes-model-node/6.4.0, Apache-2.0, approved, #6595
maven/mavencentral/io.fabric8/kubernetes-model-policy/6.4.0, Apache-2.0, approved, #6573
maven/mavencentral/io.fabric8/kubernetes-model-rbac/6.4.0, Apache-2.0, approved, #6588
maven/mavencentral/io.fabric8/kubernetes-model-scheduling/6.4.0, Apache-2.0, approved, #6610
maven/mavencentral/io.fabric8/kubernetes-model-storageclass/6.4.0, Apache-2.0, approved, #6589
maven/mavencentral/io.fabric8/kubernetes-server-mock/6.4.0, Apache-2.0, approved, #6608
maven/mavencentral/io.fabric8/mockwebserver/0.2.2, Apache-2.0, approved, clearlydefined
maven/mavencentral/io.fabric8/openshift-client-api/6.4.0, Apache-2.0, approved, #6576
maven/mavencentral/io.fabric8/openshift-client/6.4.0, Apache-2.0, approved, #6572
maven/mavencentral/io.fabric8/openshift-model-clusterautoscaling/6.4.0, Apache-2.0, approved, #6606
maven/mavencentral/io.fabric8/openshift-model-config/6.4.0, Apache-2.0, approved, #6593
maven/mavencentral/io.fabric8/openshift-model-console/6.4.0, Apache-2.0, approved, #6587
maven/mavencentral/io.fabric8/openshift-model-hive/6.4.0, Apache-2.0, approved, #6583
maven/mavencentral/io.fabric8/openshift-model-installer/6.4.0, Apache-2.0, approved, #6596
maven/mavencentral/io.fabric8/openshift-model-machine/6.4.0, Apache-2.0, approved, #6611
maven/mavencentral/io.fabric8/openshift-model-machineconfig/6.4.0, Apache-2.0, approved, #6597
maven/mavencentral/io.fabric8/openshift-model-miscellaneous/6.4.0, Apache-2.0, approved, #6571
maven/mavencentral/io.fabric8/openshift-model-monitoring/6.4.0, Apache-2.0, approved, #6585
maven/mavencentral/io.fabric8/openshift-model-operator/6.4.0, Apache-2.0, approved, #6590
maven/mavencentral/io.fabric8/openshift-model-operatorhub/6.4.0, Apache-2.0, approved, #6582
maven/mavencentral/io.fabric8/openshift-model-storageversionmigrator/6.4.0, Apache-2.0, approved, #6605
maven/mavencentral/io.fabric8/openshift-model-tuned/6.4.0, Apache-2.0, approved, #6594
maven/mavencentral/io.fabric8/openshift-model-whereabouts/6.4.0, Apache-2.0, approved, #6591
maven/mavencentral/io.fabric8/openshift-model/6.4.0, Apache-2.0, approved, #6612
maven/mavencentral/io.fabric8/openshift-server-mock/6.4.0, Apache-2.0, approved, #6579
maven/mavencentral/io.fabric8/servicecatalog-client/6.4.0, Apache-2.0, approved, #6609
maven/mavencentral/io.fabric8/servicecatalog-model/6.4.0, Apache-2.0, approved, #6575
```

File: jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/openshift/OpenshiftBuildServiceIntegrationTest.java [hunk 1]
```diff
@@ -198,7 +187,7 @@ void successfulBuild() throws Exception {
     assertThat(mockServer.getRequestCount()).isGreaterThan(8);
     collector.assertEventsRecordedInOrder("build-config-check", "new-build-config", "pushed");
     assertThat(collector.getBodies().get(1))
-            .isEqualTo("{\"apiVersion\":\"build.openshift.io/v1\",\"kind\":\"BuildConfig\",\"metadata\":{\"name\":\"myapp-s2i-suffix2\",\"namespace\":\"ns1\"},\"spec\":{\"output\":{\"to\":{\"kind\":\"ImageStreamTag\",\"name\":\"myapp:latest\"}},\"source\":{\"type\":\"Binary\"},\"strategy\":{\"sourceStrategy\":{\"forcePull\":false,\"from\":{\"kind\":\"DockerImage\",\"name\":\"myapp\"}},\"type\":\"Source\"}}}");
+            .isEqualTo("{\"apiVersion\":\"build.openshift.io/v1\",\"kind\":\"BuildConfig\",\"metadata\":{\"name\":\"myapp-s2i-suffix2\"},\"spec\":{\"output\":{\"to\":{\"kind\":\"ImageStreamTag\",\"name\":\"myapp:latest\"}},\"source\":{\"type\":\"Binary\"},\"strategy\":{\"sourceStrategy\":{\"forcePull\":false,\"from\":{\"kind\":\"DockerImage\",\"name\":\"myapp\"}},\"type\":\"Source\"}}}");
```
Comment (rohanKanojia) [inline jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/openshift/OpenshiftBuildServiceIntegrationTest.java:190]: why was namespace removed from expectations?

Comment (manusa) [reply jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/openshift/OpenshiftBuildServiceIntegrationTest.java:190]: Previously it was added automatically by the Kubernetes Client when not provided. I'll try to find the related PR for completeness.

File: jkube-kit/config/resource/src/test/java/org/eclipse/jkube/kit/config/access/ClusterAccessTest.java [hunk 2]
```diff
@@ -13,101 +13,84 @@
...
-      final OpenShiftClient oc = mock(OpenShiftClient.class);
-      when(mock.adapt(OpenShiftClient.class)).thenReturn(oc);
-      onInstantiate.accept(oc);
-    });
-  }
-
-  @AfterEach
-  void tearDown() {
-    kubernetesClientMockedConstruction.close();
+    clusterConfiguration = ClusterConfiguration.builder()
```
Comment (sunix) [inline jkube-kit/config/resource/src/test/java/org/eclipse/jkube/kit/config/access/ClusterAccessTest.java:43]: it may be good to explain in the PR description why these changes were made and any reference to a kubernetes-client PR/issue

File: jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/openshift/OpenshiftBuildServiceIntegrationTest.java (hunk 1 above)
Comment (manusa) [reply jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/openshift/OpenshiftBuildServiceIntegrationTest.java:190]: https://github.com/fabric8io/kubernetes-client/pull/4593

File: jkube-kit/config/resource/src/test/java/org/eclipse/jkube/kit/config/access/ClusterAccessTest.java (hunk 2 above)
Comment (manusa) [reply jkube-kit/config/resource/src/test/java/org/eclipse/jkube/kit/config/access/ClusterAccessTest.java:43]: These changes are just to improve the test and make use of the Kubernetes Mock Server provided by the client instead of relying on complex Mocks.

Why now? The Kubernetes Mock server as become more stable and performant over the last couple of years. Being able to test `ClusterAccess` against a semi-real cluster is easier to maintain than relying on Mocks that need extra-maintenance. In this case, the Client is slowly moving to abstract away explicit KubernetesClient implementations and moving on to SPI provided and more generic ones instead.

**Review by rohanKanojia — APPROVED** [review]

File: jkube-kit/config/resource/src/test/java/org/eclipse/jkube/kit/config/access/ClusterAccessTest.java (hunk 2 above)
Comment (sunix) [reply jkube-kit/config/resource/src/test/java/org/eclipse/jkube/kit/config/access/ClusterAccessTest.java:43]: this is good for me, ideally in the description or in another PR

**Review by sunix — APPROVED** [review]

**Comment by sunix** [comment]
This should also fix https://github.com/eclipse/jkube/issues/1540
